sys/net/gnrc/transport_layer/tcp/internal/common.h

```diff
@@ -39,14 +39,18 @@ extern "C" {
  */
 #define PORT_UNSPEC (0)
 
+/**
+ * @brief Message queue size for messaging between thread contexts
+ */
+#define TCP_MSG_QUEUE_SIZE (8U)
```

Comment:
Why did you remove the configurability of the message size(s) I would rather split it up into the different modes the `mbox` is used (open, send, recv if I see correctly) and provide a config value for each. Also, since they are stored on stack now, we should rethink if a default of 8 is not maybe to large.

Reply:
I move the define into the complation unit gnrc_tcp.c because this is the only location were the define is in use. 
I'll agree with you that splitting the defines is a good Idea.

Reply:
For me this is less about usage as about configurability. Would be nice to still have these queue size configurable as they _might_ have an impact on performance.

Reply:
In terms of configurability:
Should i move it to the public api "sys/include/net/gnrc/transport_layer/tcp/config.h" and add kconfig support then?

Reply:
Yes please, otherwise people won't find it ;-).

Reply:
Okay, I'll move the define and  setup it to four. I will not introduct a second define for a larger mbox in this PR.

The second define will be added then adding gnrc_tcp_accept, this is the only function thats requires a larger mbox.

Reply:
While I am at it:  What about making the eventloop message queue configurable?

Reply:
It isn't yet :eyes:?

Reply:
I am pretty sure it is not configureable externally.

Reply:
Then it should be :-)